TypeScript/TypeScript-React/src/App.tsx

```diff
@@ -1,15 +1,83 @@
 import './App.css'
+import { useEffect, useState } from 'react'
 import { MemoryCard } from './components/MemoryCard'
 import { itemList } from './data/items'
+import { Shuffle } from './helpers/shuffle'
+import { Item } from './ts/item.interface'
 
 export const App = () => {
+  const [items, setItems] = useState<Item[]>(
+    () => Shuffle(itemList)
+  )
+  const [openCards, setOpenCards] = useState<number[]>([]);
+  const [doneCards, setDoneCards] = useState<any>({});
+  const [clickable, setClickable] = useState<boolean>(true);
+
+  const checkForMatch = () => {
+    const [firstCard, secondCard] = openCards;
+
+    if (items[firstCard].name == items[secondCard].name) {
+      let cardName = items[firstCard].name;
+      setOpenCards([]);
+      setDoneCards((previousCard: Object) => ({
+        ...previousCard,
+        [cardName]: true
+      }));
+      setClickable(true);
+      console.log('It matched !');
+
+    } else {
+      setOpenCards([]);
+      setClickable(true);
+    }
+  }
+
+  const cardClicked = (idx: number) => {
```

Comment:
If I click the card two time it open the one is matching

Reply:
try to click the card two time and you will see that auto open the matching card

Reply:
I thing is 3 clicks to the same card when is open that open the hidden one resulting in match

Reply:
Yeah you're right, missing conditional to check clicked card index similarity. 
I'm fix it with new commits. Please have a look

[Fix Commit](https://github.com/GeraAlcantara/cards-memory-game/pull/24/commits/2076d7cb4af84d071035f58bc799af2ca25eca43)

Reply:
Look great Thank you!  🌶️